libs/utils/energy_model.py

```diff
@@ -897,6 +992,49 @@ def read_idle_states(cpu, domain_level):
                    root_power_domain=root_pd,
                    freq_domains=freq_domains)
 
+    @classmethod
+    def from_target(cls, target):
+        """
+        Create an EnergyModel by reading a target filesystem
+
+        If present, load an EM provided via dt using from_sd_target, since these
+        devices make the full EM available via the sched domain in sysfs. If
+        there is no EM at this location, attempt to load the simplified EM
+        made available via dedicated sysfs files.
+
+        :param target: Devlib target object to read filesystem from. Must have
+                       cpufreq and cpuidle modules enabled.
+        :returns: Constructed EnergyModel object based on the parameters
+                  reported by the target.
+        """
+        _log = logging.getLogger('EMReader')
+
+        def em_present_in_sd(target):
+            try:
+                cpu = target.list_online_cpus()[0]
+                d = '/proc/sys/kernel/sched_domain/cpu{}/domain0/group0/energy'.format(cpu)
+                _log.info('Checking if directory {} exists'.format(d))
+                return target.directory_exists(d)
+            except:
```

Comment:
except should match a specific exception and be restricted to the least amount of code like:
```
+            try:
+                cpu = target.list_online_cpus()[0]
+            except IndexError:  
+                return False
+            else:
+                d = '/proc/sys/kernel/sched_domain/cpu{}/domain0/group0/energy'.format(cpu)
+                _log.info('Checking if directory {} exists'.format(d))
+                return target.directory_exists(d)
```

Or even:

```
+            online_cpus = target.list_online_cpus()
+            if online_cpus:
+                cpu = online_cpus[0]
+                d = '/proc/sys/kernel/sched_domain/cpu{}/domain0/group0/energy'.format(cpu)
+                _log.info('Checking if directory {} exists'.format(d))
+                return target.directory_exists(d)
+            else:  
+                return False
```
```

Reply:
I want to catch any exception - I don't want anything which might cause an exception in one check to prevent us from doing other checks to see which EM should be loaded,